cmd/kubectl-gadget/utils/flags.go

```diff
@@ -191,19 +200,38 @@ func AddCommonFlags(command *cobra.Command, params *CommonFlags) {
 		fmt.Sprintf("Output format (%s).", strings.Join(commonutils.SupportedOutputModes, ", ")),
 	)
 
-	command.PersistentFlags().BoolVarP(
+	command.PersistentFlags().BoolVar(
 		&params.Verbose,
 		"verbose",
-		"",
 		false,
 		"Print additional information",
 	)
 
-	command.PersistentFlags().IntVarP(
+	command.PersistentFlags().IntVar(
 		&params.Timeout,
 		"timeout",
-		"",
 		0,
 		"Number of seconds that the gadget will run for",
 	)
+
+	command.PersistentFlags().StringVar(
```

Comment:
I wonder if it is also necessary to define these flags for each gadget, or if it is ok just for the `deploy` command.

Reply:
Yep. It makes only sense for the `deploy` command

Reply:
What do you mean for each gadget? 

I think those are needed for each gadget in local-gadget, but not for Inspektor-Gadget in those case the path should already be configured when deploying it. 

Also, are command line parameters the best UI here? It seems cumbersome to have to specify them for each invocation of local-gadget:

```
$ local-gadget snapshot process --containerd-socketpath=/bar/foo
$ local-gadget trace exec --containerd-socketpath=/bar/foo
```

Perhaps those should be defined as env variables too? (similar to how kubectl handles kubeconfig) (In another PR for sure). 

```
export CONTAINERD_SOCKETPATH=/bar/foo
$ local-gadget snapshot process 
$ local-gadget trace exec 
```

It seems it's possible by using viper + cobra https://github.com/kubereboot/kured/pull/464/files.

Reply:
> What do you mean for each gadget?
>
If, as for local-gadget, we want to provide the possibility of using a different path each time you call a gadget:

```bash
$ kubectl gadget trace exec --containerd-socketpath=/bar/foo
```

I proposed to add them only for the `kubectl gadget deploy` command, and I see you agree.

On the other hand, I created https://github.com/inspektor-gadget/inspektor-gadget/issues/1044 to track this improvement:

> It seems it's possible by using viper + cobra https://github.com/kubereboot/kured/pull/464/files.